apis/v1beta1/shared_types.go

```diff
@@ -75,6 +85,10 @@ type ParentReference struct {
 	// are specified, the name and port of the selected listener must match
 	// both specified values.
 	//
+	// * Service: Named port. When both Port (experimental) and SectionName
```

Comment:
Tiny nit: I think this might be a bit clearer.

```suggestion
	// * Service: Port Name. When both Port (experimental) and SectionName
```